src/compiler/factory/emitHelpers.ts

```diff
@@ -1422,6 +1438,29 @@ const disposeResourcesHelper: UnscopedEmitHelper = {
         });`,
 };
 
+const rewriteRelativeImportExtensionsHelper: UnscopedEmitHelper = {
+    name: "typescript:rewriteRelativeImportExtensions",
+    importName: "__rewriteRelativeImportExtension",
+    scoped: false,
+    text: `
+        var __rewriteRelativeImportExtension = (this && this.__rewriteRelativeImportExtension) || function (path, preserveJsx) {
+            if (typeof path === "string" && path[0] === "." && (path[1] === "/" || path[1] === "." && path[2] === "/")) {
+                if (path.substring(path.length - 4) === ".tsx") {

```

Comment:
I'm going to regret asking this, but does this (and the rest of the extension comparisons) need to be case-insensitive? At least on windows, `.TSX` and `.tsx` files are going to be the same - I dunno how loose we are on case-sensitive filesystems.